systemvm/patches/debian/config/opt/cloud/bin/cs/CsDhcp.py

```diff
@@ -54,7 +54,8 @@ def process(self):
         self.cloud.commit()
 
         # We restart DNSMASQ every time the configure.py is called in order to avoid lease problems.
-        CsHelper.service("dnsmasq", "restart")
+        if not self.cl.is_redundant() or self.cl.is_master():
```

Comment:
@jburwell Whilst that check might be convenient in some cases, I think that convenience would be a curse. For example "bring up dnsmasq on non-redundant or master routers" is a nice explicit declaration of behaviour. Simplifying that to "bring up dnsmasq on master routers" might do the same thing but it increases the learning curve for someone reading the code for the first time.